packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/src/EventGridClient.ts

```diff
@@ -25,18 +25,21 @@ import {
   RejectCloudEventsOptions,
 } from "./api/index.js";
 
+export { EventGridClientOptions } from "./api/EventGridContext.js";
+
 export class EventGridClient {
   private _client: EventGridContext;
 
   /** Azure Messaging EventGrid Client */
   constructor(
     endpoint: string,
     credential: KeyCredential,
-    options: ClientOptions = {}
+    options: EventGridClientOptions = {}
```

Comment:
Just a reminder, may need to check the ClientOptions in the rest level. Also, should we generate EventGridClientOptions if there's no actual difference between EventGridClientOptions and ClientOptions ? For RLC, the answer is no, right now.

Reply:
Let's add the logic to remove if no client level parameters in a separate PR